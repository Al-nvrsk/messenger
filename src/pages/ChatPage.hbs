{{!< chatwindow}}

<div class="ChatPage">

    <div class="userList" >
        <a href="UserInfoPage.hbs">Profile</a>
        {{>"UIComponent/inputform/inputAuth" 
            name="search"
            type="search"}}
        <div></div>
    </div>

    <div class="chatPart">chatPart
        <div class="chatPartMessage">
            <h2>Select user for chating...</h2>
        </div>
    </div>
    
</div>