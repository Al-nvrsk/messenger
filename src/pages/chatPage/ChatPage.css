.ChatPage {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}

.userList {
  border-right: 1px solid var(--main-border);
  height: 100vh;
  width: 25%;
  box-sizing: border-box;
  overflow-x: unset;
}

.userListBar {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  box-sizing: border-box;
}

.userListLink {
  color: var(--color-reject);
  margin: 20px;
  margin-bottom: 0;
  padding-bottom: 0%;
  display: flex;
  justify-content: end;
  cursor: pointer;
}

.userListLink:hover {
  color: var(--font-hover);
}

.chatPart {
  border-left: 1px solid var(--main-border);
  height: 100vh;
  width: 74%;
  box-sizing: border-box;
}

.chatPartMessage {
  box-sizing: border-box;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
}

.logoutButton {
  display: flex;
}

.mainChatting {
  box-sizing: border-box;
  display: flex;
  flex-flow: column wrap;
  width: 95%;
}

.buttonBar {
  box-sizing: border-box;
  display: flex;
  flex-flow: row nowrap;
  width: 100%;
  margin-left: 20px;
}

.messageDisplay {
  display: flex;
  height: 70%;
}

.inputMessageConteiner {
  box-sizing: border-box;
  display: flex;
  flex-flow: row nowrap;
  margin: 10px 5px;
  width: 100%;
}

.inputSubmit {
  display: flex;
  flex-flow: row nowrap;
}

.inputMessage {
  display: flex;
  color: var(--font-hover);
  flex-direction: column;
  margin: 5px 20px;
  width: 90%;
  height: 45px;
  background-color: var(--main-background);
  font-size: 18px;
  border: none;
  outline-offset: 0;
  outline: none;
  box-sizing: border-box;
}

.InfoChat {
  margin: 10px 20px;
}

.messageCardConteiner {
  display: flex;
  flex-direction: column-reverse;
  align-items: flex-start;
  height: 65vh;
  overflow: hidden;
  overflow-y: auto;
}

.emptyField {
  display: flex;
  height: 100vh;
  font-size: 20px;
  align-items: center;
  justify-content: center;
}
