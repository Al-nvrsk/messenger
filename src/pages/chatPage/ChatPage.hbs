<main>
<div class = "ChatPage">

    <div class = "userList" >
        <a href = "UserInfoPage.hbs"> Profile </a>
        {{> "UIComponent/inputForm/inputAuth" 
            name = "search"
            type = "search"}}
    </div>

    <div class = "chatPart"> 
        <div class = "chatPartMessage">
            <h2> Select user for chating... </h2>
        </div>
    </div>
    
</div>
</main>
