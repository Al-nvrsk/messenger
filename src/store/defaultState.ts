export interface AppState {
  appIsInited: boolean
  isLoading: boolean
  isAuth: boolean
  loginFormError: string | null
  user: User | null
  chats: object[] | null
}

export const defaultState: AppState = {
  appIsInited: false,
  isLoading: false,
  isAuth: false,
  loginFormError: null,
  user: null,
  chats: null
}
