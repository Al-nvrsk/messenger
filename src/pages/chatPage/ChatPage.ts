import Block from 'utils/Block'
import './ChatPage.css'

export default class ChatPage extends Block {
  static componentName = 'ChatPage'
  render (): string {
    return `
      <main>
        <div class = "ChatPage">

          <div class = "userList" >
            <a href = "UserInfoPage.hbs"> Profile </a>
            {{{ InputSearch 
            name = "search"
            type = "search"
            placeholder = "search"}}}
          </div>

          <div class = "chatPart"> 
            <div class = "chatPartMessage">
              <h2> Select user for chating... </h2>
            </div>
          </div>
    
        </div>
      </main>
    `
  }
}
