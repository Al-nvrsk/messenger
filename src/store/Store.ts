import set from 'utils/helper/set'
import EventBus from 'utils/EventBus'
import { defaultState } from './defaultState'

export enum StoreEvents {
  Updated = 'updated',
  Deleted = 'deleted'
}

class Store extends EventBus {
  private readonly state: Indexed = defaultState

  public getState (): Indexed {
    return this.state
  }

  public setState (path: string, value: unknown): void {
    set(this.state, path, value)
    this.emit(StoreEvents.Updated)
  };
}

export default new Store()
