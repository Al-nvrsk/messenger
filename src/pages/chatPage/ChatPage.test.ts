import '@testing-library/jest-dom'
import ChatPage from './ChatPage'
import store from 'store/Store'
import mockedRenderDOM from 'test/mockedRenderDOM'
import mockedRegisteredComp from 'test/mockedRegisteredComp'

jest.mock('utils/renderDOM')

describe('pages/ChatPage', () => {
  it('should render ChatPage', () => {
    store.setState('isAuth', true)
    mockedRegisteredComp()
    mockedRenderDOM(new ChatPage(store))
    const chatPageBody = document.querySelector('.ChatPage')
    expect(chatPageBody).toMatchSnapshot()
  })
})
