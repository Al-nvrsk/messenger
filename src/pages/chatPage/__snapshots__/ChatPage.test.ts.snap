// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/ChatPage should render ChatPage 1`] = `
<div
  class="ChatPage"
>
  
            
  <div
    class="userList"
  >
    
              
    <div
      class="userListBar"
    >
      
                
      <button
        class="buttonChange"
        type="button"
      >
         Create new chat 
      </button>
      
                
      <button
        class="buttonChange"
        type="button"
      >
         UserInfo 
      </button>
      
              
    </div>
      
              
    <div
      class="inputAuth"
    >
      
        
      <label
        class="inputAuthLabel"
        for="search"
      >
        
          
        <b>
            
        </b>
        
        
      </label>
      
        
      <input
        class="inputAuthInput"
        name="search"
        placeholder="search"
        required=""
        type="search"
      />
      
        
      <hr
        class="inputAuthHr"
      />
       
      
    </div>
    
              undefined
            
  </div>
  
            
  <div
    class="chatPart"
  >
     
              
    <div
      class="logoutButton"
    >
      
                
      <button
        class="buttonChange"
        type="button"
      >
         Logout 
      </button>
      
              
    </div>
    
              
    <div
      class="emptyField"
    >
      
                  Select Chat for chating...
                
    </div>
    
            
  </div>
  
          
</div>
`;
